catalyst/loggers/comet.py

```diff
@@ -0,0 +1,264 @@
+from typing import Dict, List
+import pickle
+
+import numpy as np
+
+from catalyst.core.logger import ILogger
+from catalyst.settings import SETTINGS
+
+if SETTINGS.comet_required:
+    import comet_ml
+
+
+class CometLogger(ILogger):
+    """Comet logger for parameters, metrics, images and other artifacts
+    (videos, audio, model checkpoints, etc.).
+
+    You will need a Comet API Key to log your Catalyst runs to Comet.
+    You can sign up for a free account here: https://www.comet.ml/signup
+
+    Check out our Quickstart Guide to learn more:
+    https://www.comet.ml/docs/quick-start/.
+
+    Args:
+        workspace : Optional,``str``, Workspace used to log the experiment.
+        project_name: Optional,``str``, Project used to log the experiment.
+        experiment_id: Optional,``str``, Experiment Key string of an existing Experiment.
+            Used to continue logging to an existing experiment (resume experiment).
+        tags: Optional,``list[str]``, A list of tags to add to the Experiment.
+        experiment_kwargs: Optional,``dict``, Additional keyword arguments.
+            Used to pass additional arguments to the Experiment object
+
+    Python API examples:
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        runner = dl.SupervisedRunner()
+        runner.train(
+            ...
+            loggers={
+                "comet": dl.CometLogger(
+                    project_name="my-comet-project
+                )
+            }
+        )
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        class CustomRunner(dl.IRunner):
+            # ...
+
+            def get_loggers(self):
+                return {
+                    "console": dl.ConsoleLogger(),
+                    "comet": dl.CometLogger(
+                        project_name="my-comet-project"
+                    )
+                }
+            # ...
+
+        runner = CustomRunner().run()
+
+    Config API example:
+    .. code-block:: yaml
+        loggers:
+            neptune:
+                _target_: CometLogger
+                project_name: my_project
+        ...
+    Hydra API example:
+    .. code-block:: yaml
+        loggers:
+            neptune:
+                _target_: catalyst.dl.CometLogger
+                project_name: my_project
+        ...
+    """
+
+    def __init__(
+        self,
+        workspace: str = None,
+        project_name: str = None,
+        experiment_id: str = None,
+        comet_mode: str = "online",
+        tags: List = None,
+        logging_frequency: int = 1,
+        **experiment_kwargs,
```

Comment:
mb add `api_key` as an explicit parameter?

Reply:
Hi @Podidiving, generally we recommend users set their API Key either as an environment variable or in a `comet.config` file on their machine.